lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -311,3 +313,50 @@ def test_get_due_dates_error_fetching_block(self):
                 ]
                 actual_due_dates = get_due_dates(request, self.course.id, self.staff_user)
                 assert expected_due_dates == actual_due_dates
+
+
+class CourseKeyListTestMixin(CourseApiTestMixin):
+    """
+    Common behavior for list_course_keys tests
+    """
+
+    def _make_api_call(self, requesting_user, specified_user, role=None):
+        """
+        Call the list_course_keys api endpoint to get information about
+        `specified_user` on behalf of `requesting_user`.
+        """
+        request = Request(self.request_factory.get('/'))
+        request.user = requesting_user
+        with check_mongo_calls(0):
+            return list_course_keys(request, specified_user.username, role=role)
+
+    def verify_courses(self, courses):
+        """
+        Verify that there is one course, and that it has the expected format.
+        """
+        self.assertEqual(len(courses), 1)
+        self.verify_course(courses[0])
+
+
+class TestGetCourseKeyList(CourseKeyListTestMixin, SharedModuleStoreTestCase):
+    """
+    Test the behavior of the `list_course_keys` api function.
+    """
+    ENABLED_SIGNALS = ['course_published']
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestGetCourseKeyList, cls).setUpClass()
+        cls.toy_course = CourseKey.from_string(u'edX/toy/2012_Fall')
+
+        # Make a test user and give it staff level access to the toy course:
+        cls.course_staff_user = cls.create_user("course_staff_user")
+        add_users(self.global_admin, CourseStaffRole(cls.toy_course.id), cls.course_staff_user)
+
+        # Construct some extra courses, for noise:
+        cls.extra_course1 = cls.create_course()
+        cls.extra_course2 = cls.create_course()
+
+    def test_as_course_staff(self):
```

Comment:
This obviously doesn't cover all the cases, but I only wrote it out of desperation because the only other test that covers this function is higher level (at the view-level) and it was failing without enough explanation.

Reply:
So if it passes CI, do we get rid of this test?

Reply:
I did just that.